src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/StripEmbeddedLibraries.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Mono.Cecil;
+using Mono.Linker;
+using Mono.Linker.Steps;
+using System;
+using System.Linq;
+using Xamarin.Android.Tasks;
+
+namespace MonoDroid.Tuner
+{
+	public class StripEmbeddedLibraries : BaseStep
+	{
+		protected override void ProcessAssembly (AssemblyDefinition assembly)
+		{
+			if (!Annotations.HasAction (assembly))
+				return;
+
+			var action = Annotations.GetAction (assembly);
+			if (action == AssemblyAction.Skip || action == AssemblyAction.Delete)
+				return;
+
+			var fileName = assembly.Name.Name + ".dll";
+			if (MonoAndroidHelper.IsFrameworkAssembly (fileName) && !MonoAndroidHelper.FrameworkEmbeddedJarLookupTargets.Contains (fileName))
+				return;
+
+			bool assembly_modified = false;
+			foreach (var mod in assembly.Modules) {
+				// embedded jars
+				var resjars = mod.Resources.Where (r => r.Name.EndsWith (".jar", StringComparison.InvariantCultureIgnoreCase)).Select (r => (EmbeddedResource)r);
+				foreach (var resjar in resjars.ToArray ()) {
+					Context.LogMessage ("    Stripped {0} from {1}", resjar.Name, fileName);
+					mod.Resources.Remove (resjar);
+					assembly_modified = true;
+				}
+				// embedded AndroidNativeLibrary archive
+				var nativezip = mod.Resources.FirstOrDefault (r => r.Name == "__AndroidNativeLibraries__.zip") as EmbeddedResource;
+				if (nativezip != null) {
+					Context.LogMessage ("    Stripped {0} from {1}", nativezip.Name, fileName);
+					mod.Resources.Remove (nativezip);
+					assembly_modified = true;
+				}
+				// embedded AndroidResourceLibrary archive
+				var reszip = mod.Resources.FirstOrDefault (r => r.Name == "__AndroidLibraryProjects__.zip") as EmbeddedResource;
+				if (reszip != null) {
+					Context.LogMessage ("    Stripped {0} from {1}", reszip.Name, fileName);
+					mod.Resources.Remove (reszip);
+					assembly_modified = true;
+				}
+			}
+			if (assembly_modified && action == AssemblyAction.Copy) {
```

Comment:
Once we process `AssemblyAction.Skip` assemblies, this would need to be updated.